### PR TITLE
refactor: [DRAFT] misc logic and type-safety cleanups

### PR DESCRIPTION
**Pull request by beyackle**
## Description

Some refactoring to make the logic clearer in places and to tweak efficiency a little.

## Task Item

#minor

### CODE REVIEW

File: Composer/packages/electron-server/src/electronWindow.ts [hunk 1]
```diff
@@ -54,7 +54,7 @@ export default class ElectronWindow {
   }
 
   public static get isBrowserWindowCreated() {
-    return !!ElectronWindow.instance;
+    return ElectronWindow.instance != null;
```
Comment (a-b-r-o-w-n) [inline Composer/packages/electron-server/src/electronWindow.ts:57]: triple equals here please.

File: Composer/packages/extensions/extension/src/hooks/useLgApi.ts [hunk 2]
```diff
@@ -12,10 +12,9 @@ export const useLgApi = (shellApi: ShellApi) => {
   const deleteLgTemplates = (lgFileId: string, lgTemplates: string[]) => {
     const normalizedLgTemplates = lgTemplates
       .map((x) => {
-        const lgTemplateRef = LgTemplateRef.parse(x);
-        return lgTemplateRef ? lgTemplateRef.name : '';
+        return LgTemplateRef.parse(x)?.name;
       })
-      .filter((x) => !!x);
+      .filter((x) => x != null) as string[];
```
Comment (a-b-r-o-w-n) [inline Composer/packages/extensions/extension/src/hooks/useLgApi.ts:17]: This is technically not the same behavior. 

I would suggest something like `filter(Boolean)`

File: Composer/packages/lib/indexers/__tests__/utils/diagnosticUtil.test.ts [hunk 3]
```diff
@@ -1,42 +1,42 @@
...
 
-import { createSingleMessage, combineMessage, findErrors, isValid } from '../../src/utils';
+import { createSingleMessage, combineMessage, findProblems, isValid } from '../../src/utils';
 
 const diagnostics = [
   {
     message: 'syntax error',
     range: { start: { line: 15, character: 0 }, end: { line: 15, character: 1 } },
     severity: 0,
-    source: 'a.lu',
```
Comment (a-b-r-o-w-n) [inline Composer/packages/lib/indexers/__tests__/utils/diagnosticUtil.test.ts:11]: Is this intentional? I thought prettier required trailing commas.

File: Composer/packages/lib/indexers/src/dialogUtils/dialogChecker.ts [hunk 4]
```diff
@@ -27,19 +27,19 @@ export const createPath = (path: string, type: string): string => {
   return `${list[0]}${focused}#${type}#${list[1]}`;
 };
 
-function findAllRequiredProperties(schema: any): { [key: string]: boolean } {
-  if (!schema) return {};
+function findAllRequiredProperties(schema: any): Set<string> {
```
Comment (a-b-r-o-w-n) [inline Composer/packages/lib/indexers/src/dialogUtils/dialogChecker.ts:30]: :+1:

File: Composer/packages/lib/code-editor/src/BaseEditor.tsx [hunk 5]
```diff
@@ -182,18 +182,16 @@ const BaseEditor: React.FC<BaseEditorProps> = (props) => {
...
   }, [diagnostics]);
 
-  const warningMsgFromDiagnostics = useMemo(() => {
-    const warnings = findWarnings(diagnostics);
-    return warnings.length ? combineSimpleMessage(warnings) : '';
-  }, [diagnostics]);
-
-  const hasError = !!errorMessage || !!errorMsgFromDiagnostics;
-  const hasWarning = !!warningMessage || !!warningMsgFromDiagnostics;
+  const hasError = errorMessage ?? errorMsgFromDiagnostics;
```
Comment (srinaath) [inline Composer/packages/lib/code-editor/src/BaseEditor.tsx:193]: From the variable name it looks like its expecting a boolean right?
shouldn't this be !!(errorMessage || warningMsgFromDiagnostics).

For ex
'abc' ?? undefined = 'abc'
!!('abc' || undefined) = true - We want the value to be represented as boolean.

Am i missing something here?

**Comment by beyackle** [comment]
Closing in favor of the later ones that do this in a more broken-down fashion.

**Pull request closed without merging**